### ISSUE TITLE
Kinematic Rigid Bodies do not update their position with the world despite calling `update_kinematic_position`

### ISSUE THREAD

**Issue by samaursa** [xissburg/edyn#141]
I have tested the following successfully:
- 2 rigidbodies moving towards each other and generating accurate contacts
- 1 rigidbody moving towards a non-moving (but not static) kinematic body and generating accurate contacts

If I move the kinematic RigidBody using `update_kinematic_position`, the contact points generated are as if the kinematic RigidBody never moved i.e. their position is that of when the kinematic RigidBody was first created. In effect, it seems that kinematic RigidBodies cannot be moved at all.

Perhaps there is another call that needs to be made per entity that is a kinematic RB to update the position with the world that I don't know of?

**Comment by xissburg** [comment]
Thanks for bringing this up! These functions have been neglected for quite a while and need to be revisited or perhaps even removed to avoid confusion. The reason they're not working properly though is because they're not calling `registry.patch` after modifying components. For the time being, the correct way to move a kinematic is by assigning position and orientation and also the velocity. See [`ExamplePlatforms::updatePhysics`](https://github.com/xissburg/edyn-testbed/blob/96d428f17c3cb71dddd4a622b625e140c1be543e/client/src/platforms.cpp#L57).

**Comment by samaursa** [comment]
Thanks @xissburg I'll try that out

> or perhaps even removed to avoid confusion

If they are removed, how do we move the Kinematic bodies without these functions?

**Comment by xissburg** [comment]
> Edit: oh I see... do you mean we should patch the position/orientation and velocity directly?

Exactly. It's important to set the appropriate angular and linear velocity for constraints to behave correctly.

**Comment by samaursa** [comment]
@xissburg since I am using the `update_kinematic_position` function, I updated it to test whether the overlaps will work or not:
```
void update_kinematic_position(entt::registry &registry, entt::entity entity, const vector3 &pos, scalar dt) {
    EDYN_ASSERT(registry.any_of<kinematic_tag>(entity));
    auto &curpos = registry.get<position>(entity);
    auto &vel = registry.get<linvel>(entity);
    vel = (pos - curpos) / dt;
    registry.patch<position>(entity) = pos;
    registry.patch<linvel>(entity) = vel;
    registry.patch<angvel>(entity) = registry.get<angvel>(entity);
    registry.patch<orientation>(entity) = registry.get<orientation>(entity);
    curpos = pos;
}
```
However, these changes do not seem to be working as expected.
